website/src/main/tut/docs/rules/FixSyntax.md

```diff
@@ -0,0 +1,26 @@
+---
+layout: docs
+title: FixSyntax
+---
+
+# FixSyntax
+
+_Since 0.5.8_
+
+This rule contains several sub-rules to rewrite "unsafe" syntax. 
+
+## Configuration
+
+By default, this rule does not rewrite syntax.
+
+It contains the following sub-rules:
+
+* `removeFinalVal` transforms `final val` to `val`
```

Comment:
Why `final` is not desired ? Some extra information about the problems being solved with these rules might increase the usage of this tool!

Reply:
Yeah, I should add more docs for this.

Reply:
I agree, I google it and it's not super obvious why we would want to change from `final val` to `val`. Also the change is not purely syntactic: `final val` gets constant-folded, i.e. if you have

```scala
class Foo {
  final val x = "42"
}
```

anywhere `x` appears, it's directly replaced with "42".

Removing the `final` removes this optimization, since now `x` can be overridden by subclasses.

Reply:
Ah, I've seen `FinalVal` in http://www.wartremover.org/doc/warts.html.

Apparently the constant-folding mechanisms breaks incremental compilation, so the purpose of this rule is precisely to avoid it.

Zinc bug: https://github.com/sbt/zinc/issues/227

Reply:
This definitely deserves an explanation somewhere :D